src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -800,6 +803,78 @@ process_touch_if_changed (GError **error)
   return TRUE;
 }
 
+/* This is adapted from shell_global_reexec_self() from gnome-shell */
+static gboolean
+reexec_self_mock_compat (GError **error)
```

Comment:
That whole function is really interesting!

Reply:
9 years later... https://git.gnome.org/browse/gnome-shell/commit/?id=6a1e408c44f8646b7d2a1c2bad7e14160dbeeb59

Reply:
`svn path=/trunk/; revision=130`!  Man, those were the days...